model/UserSchema.js

```diff
@@ -20,10 +20,39 @@ const userSchema = new mongooose.Schema(
       type: String,
       required: true,
     },
+    tokens:[
+      {
+          token:{
+              type:String,
+              require:true
+          }
+      }
+  ]
   },
-  { versionKey: false, timestamps: true }
 );
 
+
+//hashing the passwords with a total of 20 rounds
+userSchema.pre('save', async function(next){
+  if(this.isModified('password')) {
+    this.password = await bcrypt.hash(this.password,20);
+  }
+  next();
+})
+
+//generating token using key
+userSchema.methods.generateAuthToken = async function(){
+  try {
+    let tokenNew = jwt.sign({_id:this._id}, process.env.SECRET_KEY);
+    thus.token = this.tokens.concat({token:tokenNew});
+    await this.save();
+    return tokenNew;
+
+  } catch (error) {
+    console.log(error);
```

Comment:
consoles are good for debugging but remember to remove these when pushing to prod